test/extension/handlers/topologymutation/handler.go

```diff
@@ -313,6 +359,38 @@ func patchDockerMachineTemplate(ctx context.Context, dockerMachineTemplate *infr
 	return errors.New("no version variables found for DockerMachineTemplate patch")
 }
 
+// patchDockerMachineTemplate patches the DockerMachinePoolTemplate.
+// It sets the CustomImage to an image for the version in use by the MachinePool.
+// NOTE: this patch is not required anymore after the introduction of the kind mapper in kind, however we keep it
+// as example of version aware patches.
+func patchDockerMachinePoolTemplate(ctx context.Context, dockerMachinePoolTemplate *infraexpv1.DockerMachinePoolTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
```

Comment:
Please diff again against patchDockerMachineTemplate we changed some things there